app/assets/javascripts/browsered/web-payments/format-card-types.js

```diff
@@ -5,11 +5,10 @@ module.exports = (allowedCardTypes, provider) => {
     return []
   }
 
-  const availableNetworks = allowedCardTypes.map(type => {
-    if (type.debit || type.credit) {
-      return type.brand
-    }
-  })
+  const availableNetworks = allowedCardTypes.reduce((brands, type) => {
```

Comment:
can we not continue to use `map`?
Just to reduce the number of changes we need to make.

Reply:
Standard 16 complains about array transforms that don't return a value in all cases. `.map`, generally, is for transforming data from one array to another without skipping any elements. 

As we are only interested in elements that match our conditional we should be filtering before mapping, we can combine a `.filter` and `.map` into one by using a `.reduce` function here.